after/main.py

```diff
@@ -1,56 +1,53 @@
-import hydra
 import torch
-from hydra.core.config_store import ConfigStore
-from omegaconf import OmegaConf
+from jsonargparse import CLI
 
-from config import MNISTConfig
+import config
 from ds.dataset import create_dataloader
 from ds.models import LinearNet
 from ds.runner import Runner, run_epoch
 from ds.tracking import TensorboardExperiment
 
-cs = ConfigStore.instance()
-cs.store(name="mnist_config", node=MNISTConfig)
 
-
-@hydra.main(config_path="conf", config_name="config")
-def main(cfg: MNISTConfig) -> None:
-    print(OmegaConf.to_yaml(cfg))
+def main(
+    paths: config.Paths,
+    files: config.Files,
+    params: config.Params,
```

Comment:
In this case all parameters are dataclasses, but it can be simple parameters with their type hint. There is support for a wide range of type hints, including deeply nested and complex, see for example https://jsonargparse.readthedocs.io/en/stable/#type-hints